tiledb/sm/array_schema/dimension.h

```diff
@@ -1043,13 +1043,20 @@ class Dimension {
           "Domain check failed; Upper domain bound should "
           "not be smaller than the lower one"));
 
-    // Domain range must not exceed the maximum uint64_t number
-    // for integer domains
-    if ((uint64_t)(domain[1] - domain[0]) ==
-        std::numeric_limits<uint64_t>::max())
+    // Bounds should fall within numeric limits for given datatype
+    if (domain[0] < std::numeric_limits<T>::min() or
+        domain[0] > std::numeric_limits<T>::max()) {
```

Comment:
Neither of these comparisons can ever evaluate to 'true'. It's part of the definition of `min` and `max` for an integral type T is that there are no *representable* numbers in the type that do not fall within the closed interval with these bounds. To be certain, there are integers that are lower and higher, but they're not in type T.

As a result, this change has the effect of removing the existing conditional entirely. If that's desirable, then simply remove it.

Reply:
Thank you; I appreciate the explanation. I'll go ahead and close this PR then.